### PR TITLE
Rename SlabMemoryPool #31609

### LINKED ISSUE #31609: Rename SlabMemoryPool
The SlabMemoryPool no longer use slabs - https://github.com/dotnet/aspnetcore/pull/30732 - and should be renamed.

Possible new name: `PinnedBlockMemoryPool`

@davidfowl

### PR DESCRIPTION
**Pull request by careless6666**
Renamed the SlabMemoryPool class and related objects to the suggested name PinnedBlockMemoryPool.

Fixes https://github.com/dotnet/aspnetcore/issues/31609

### CODE REVIEW

**Comment by dnfadmin** [comment]
[CLA assistant check](https://cla.dotnetfoundation.org/dotnet/aspnetcore?pullRequest=31752) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.dotnetfoundation.org/dotnet/aspnetcore?pullRequest=31752) before we can accept your contribution.<br/><br/>:x: careless6666 [sign now](https://cla.dotnetfoundation.org/dotnet/aspnetcore?pullRequest=31752)<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.dotnetfoundation.org/check/dotnet/aspnetcore?pullRequest=31752) it.</sub>

**Review by davidfowl — APPROVED** [review]

**Review by BrennanConroy — APPROVED** [review]

File: src/Servers/Kestrel/Core/test/SlabMemoryPoolTests.cs [hunk 1]
```diff
@@ -6,9 +6,9 @@
 
 namespace Microsoft.Extensions.Internal.Test
 {
-    public  class SlabMemoryPoolTests: MemoryPoolTests
+    public  class PinnedBlockMemoryPoolTests: MemoryPoolTests
```
[the remark this answers is not in the document]
Comment (careless6666) [reply src/Servers/Kestrel/Core/test/SlabMemoryPoolTests.cs:9]: thx, renamed

[the remark this answers is not in the document]
Comment (halter73) [reply src/Servers/Kestrel/Core/test/SlabMemoryPoolTests.cs:9]: If only that could easily be done from the github ui.

File: src/Servers/Kestrel/Core/test/ConcurrentPipeWriterTests.cs [hunk 2]
```diff
@@ -16,7 +16,7 @@ public class ConcurrentPipeWriterTests
         [Fact]
         public async Task PassthroughIfAllFlushesAreAwaited()
         {
-            using (var slabPool = new SlabMemoryPool())
+            using (var slabPool = new PinnedBlockMemoryPool())
```
Comment (JamesNK) [inline src/Servers/Kestrel/Core/test/ConcurrentPipeWriterTests.cs:19]: nit: slabPool -> pool or memoryPool

Comment (JamesNK) [inline src/Servers/Kestrel/Core/test/ConcurrentPipeWriterTests.cs:19]: nit: slabPool -> memoryPool

Repeat throughout this file.

**Review by JamesNK — APPROVED** [review]

**Comment by BrennanConroy** [comment]
Thanks @careless6666 !

**Pull request merged**